### DISCUSSION TITLE
Reusing milvus will delete data

### DISCUSSION THREAD

**Discussion by sunjiaqi52777** [milvus-io/milvus#31976] [Q&A and General discussion]
I had one existed milvus cluster and wanted to create a new one.
However, while trying reusing the data from the previous milvus with same rootPath in yaml, milvus seems default erase all existed data, that cause our all data lost. Is there any document to illustrate this issue of erasing data?

**Comment by yhmo** [comment]
A single Milvus never "erase all existing data". But two milvus running on the same rootpath will clean data of each other. Never, never, put two alive milvus on the same rootpath.
Could you show me more details on how you deploy the new cluster?

**Reply by sunjiaqi52777** [reply]
Thanks for reply. When I created the new cluster, I set externalS3 configuration pointing the previous cluster and didn't change bucket name and rootPath.

> externalS3:
  enabled: true
  host: milvus-cluster-dev-minio
  port: 9000
  accessKey: minioadmin
  secretKey: minioadmin
  useSSL: false
  bucketName: milvus-bucket
  rootPath: file

**Reply by yhmo** [reply]
Milvus has two kinds of data:
- metadata, stored in Etcd, including collections schema, system runtime status, and S3 path of each segment
- data files, stored in S3, vectors are stored here, including indexes files

Milvus finds the data files according to the paths in the Etcd.
To reuse the previous cluster's data, the previous Etcd and S3 are both required.

**Reply by fangwch** [reply]
I tried the same case with standalone, and found the data is not deleted. So does milvus have different behaviour for distributed and standalone mode? @yhmo , thanks.

----
I found precheck bucket logs in the second standalone milvus (milvus-standalone-01):

**I20240408 14:23:06.352175    15 MinioChunkManager.cpp:224] [SEGCORE][PreCheck][milvus] start to precheck chunk manager with configuration:[address=10.88.3.2:9000, bucket_name=a-bucket, root_path=files, storage_type=remote, cloud_provider=aws, iam_endpoint=, log_level=fatal, region=, useSSL=false, useIAM=false, useVirtualHost=false, requestTimeoutMs=10000]**

Here are two compose files that i do experiment with:

[docker-composes.zip](https://github.com/milvus-io/milvus/files/14911751/docker-composes.zip)

Steps of experiment:

1. docker-compose -f milvus-standalone up -d
2. insert some data into milvus-standalone
3. docker-compose -f milvus-standalone-01 up -d
4. insert some data into milvus-standalone-01

**Reply by yhmo** [reply]
milvus-standalone.yaml starts 4 containers: milvus-etcd, milvus-standalone, milvus-minio, attu. The 4 containers are in the same network "default name=milvus". The milvus-minio is visible to another network "minio name=minio".

milvus-standalone-01.yaml starts 2 containers: milvus-etcd-01, milvus-standalone-01. The 2 containers are in two networks: "default name=milvus-01" and "minio name=minio".

So, there are 2 standalones, share the same milvus-minio. You didn't do any configurations to the standalones, so, they are using the same rootPath of the milvus-minio.

As I mentioned, "two milvus running on the same rootpath will clean data of each other". If one of the standalones triggers a GC action, it might clean the data of another standalone.(by default, milvus triggers a GC action for each 1~2 hours)

In my opinion, no need to do so much complicate settings. Just use different etcd, different minio for different standalone.
**A simple rule to avoid all kinds of weird problems is: do the work in a simple way.**
